pennylane_lightning/core/src/simulators/lightning_qubit/tests/Test_StateVectorLQubitManaged.cpp

```diff
@@ -107,4 +108,60 @@ TEMPLATE_TEST_CASE("StateVectorLQubitManaged::StateVectorLQubitManaged",
 
         REQUIRE(sv.getDataVector() == approx(st_data));
     }
-}
\ No newline at end of file
+}
+
+TEMPLATE_TEST_CASE("StateVectorLQubitManaged::setBasisState",
+                   "[StateVectorLQubitManaged]", float, double) {
+    using PrecisionT = TestType;
+    using ComplexT = std::complex<PrecisionT>;
+    using TestVectorT = TestVector<ComplexT>;
+
+    const std::size_t num_qubits = 3;
+
+    SECTION("Prepares a single computational basis state.") {
+        TestVectorT init_state =
+            createRandomStateVectorData<PrecisionT>(re, num_qubits);
+
+        TestVectorT expected_state(size_t{1U} << num_qubits, 0.0,
+                                   getBestAllocator<ComplexT>());
+        expected_state[3] = {1.0, 0.0};
+
+        StateVectorLQubitManaged<PrecisionT> sv(init_state);
+
+        size_t index = 3;
+        sv.setBasisState(index);
+
+        REQUIRE(sv.getDataVector() == approx(expected_state));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorLQubitManaged::SetStateVector",
+                   "[StateVectorLQubitManaged]", float, double) {
+    using PrecisionT = TestType;
+    using ComplexT = std::complex<PrecisionT>;
+    using TestVectorT = TestVector<ComplexT>;
+
+    const std::size_t num_qubits = 3;
+
+    SECTION("Set state vector with values and indices") {
+        TestVectorT init_state =
+            createRandomStateVectorData<PrecisionT>(re, num_qubits);
+
+        auto expected_state = init_state;
+
+        for (size_t i = 0; i < Pennylane::Util::exp2(num_qubits - 1); i++) {
+            std::swap(expected_state[i * 2], expected_state[i * 2 + 1]);
+        }
+
+        std::vector<std::size_t> indices = {0, 2, 4, 6, 1, 3, 5, 7};
```

Comment:
Should we also test for cases that are not permutations? For example, number of indices smaller, greater than the number of SV entries, repeated indices, etc.

Reply:
This is always a good idea, but I think it is out of scope for this work.